subdomain.tf

```diff
@@ -1,11 +1,20 @@
-resource "ns_autogen_subdomain" "subdomain" {}
+data "ns_subdomain" "subdomain" {
```

Comment:
As discussed earlier, could collapse this to `data "ns_block" "this"`. Did you want to do that in this PR or another PR?

Reply:
Yes, I'm going to do this in the next PR.